py-polars/polars/expr/datetime.py

```diff
@@ -1423,6 +1423,83 @@ def replace_time_zone(
             self._pyexpr.dt_replace_time_zone(time_zone, ambiguous._pyexpr)
         )
 
+    def to_naive_local(
+        self,
+        tz: IntoExpr,
+    ) -> Expr:
+        """
+        Convert to local datetime in given time zone.
+
+        Parameters
+        ----------
+        tz
+            Time zone for the `Datetime` expression.
+
+        Returns
+        -------
+        Expr
+            Expression of data type :class:`DateTime`.
+
+        Examples
+        --------
+        >>> from datetime import datetime
+        >>> df = pl.DataFrame(
+        ...     {
+        ...         "date_col": [datetime(2020, 10, 10)] * 3,
+        ...         "timezone": ["Europe/London", "Africa/Kigali", "America/New_York"],
+        ...     }
+        ... ).with_columns(pl.col("date_col").dt.replace_time_zone("UTC"))
+        >>> df.with_columns(
+        ...     pl.col("date_col").dt.to_naive_local("timezone").alias("local_dt")
+        ... )
+        shape: (3, 3)
+        ┌─────────────────────────┬──────────────────┬─────────────────────┐
+        │ date_col                ┆ timezone         ┆ local_dt            │
+        │ ---                     ┆ ---              ┆ ---                 │
+        │ datetime[μs, UTC]       ┆ str              ┆ datetime[μs]        │
+        ╞═════════════════════════╪══════════════════╪═════════════════════╡
+        │ 2020-10-10 00:00:00 UTC ┆ Europe/London    ┆ 2020-10-10 01:00:00 │
+        │ 2020-10-10 00:00:00 UTC ┆ Africa/Kigali    ┆ 2020-10-10 02:00:00 │
+        │ 2020-10-10 00:00:00 UTC ┆ America/New_York ┆ 2020-10-09 20:00:00 │
+        └─────────────────────────┴──────────────────┴─────────────────────┘
+
+        """
+        local_tz = parse_as_expression(tz, str_as_lit=False)
+        return wrap_expr(self._pyexpr.dt_to_naive_local(local_tz))
+
+    def from_naive_local(
+        self,
+        naive_tz: IntoExpr,
+        out_tz: str,
+        ambiguous: Ambiguous = "raise",
```

Comment:
ok for a follow-up, but this would ideally be an expression too, right?

Reply:
`try_ternary_elementwise`

Reply:
Nice, will try to build with that in the follow up :D